api/api.proto

```diff
@@ -607,6 +607,32 @@ enum NearSyncReplicationStrategy {
     NEAR_SYNC_STRATEGY_OPTIMIZED = 2;
 }
 
+enum AccessMode {
+  // maps to AccessMode of CSI Specification
+  UNKNOWN = 0;
+  SINGLE_NODE_WRITER = 1; // read write vol, attached from a single node
+  SINGLE_NODE_READER_ONLY = 2; // read only vol, attached from single node
+  MULTI_NODE_READER_ONLY = 3; // rox - read only many from many nodes
+  MULTI_NODE_SINGLE_WRITER = 4; // read many write one volume support
+  MULTI_NODE_MULTI_WRITER = 5; // unrestricted rwx
+  SINGLE_NODE_SINGLE_WRITER = 6; // read write vol, from a single node
+  SINGLE_NODE_MULTI_WRITER = 7; // px default volume support
+}
+
+// Migration object applied on volume spec tracks vol migration
+// of shared raw block volumes in kubevirt VM migration usecase.
+message Migration {
+  bool active = 1; // whether migration active
+
+  string source_address = 2; // source migration net address
+  uint32 source_node = 3; // source migration internal node id
+  string source_machine = 4; // source migration external machine id
```

Comment:
do we need both (here and below ?) Ideally you should use only machine_id

Reply:
This structure is used only for testing migration workflow.
source_machine is the external cli interface (pxctl host attach --migrate <machine id> vol)
source_node is the conversion to internal px-storage node id.
they are setup part of the spec, so grpc can push them along to px-storage. grpc does not have access to methods to do conversion.